src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -150,29 +180,37 @@ FileMeta getMetadata( final File target, final LocalLockOwner owner ) throws IOE
         {
             return lockManager.reentrantSynchronous( path, ( opLock ) -> {
 
-                // we need to use cache.getAdvancedCache().getTransactionManager() to lock this entry and then update the node-level locks.
-                // this will be where we reserve the lock level (or change it, if our node is already represented)
-                FileMeta meta = metadataCache.computeIfAbsent( path, ( p ) -> new FileMeta( p, target.isDirectory(),
-                                                                                            owner ) );
+                TransactionManager transactionManager = metadataCache.getAdvancedCache().getTransactionManager();
+                transactionManager.begin();
+                FileMeta meta = metadataCache.computeIfAbsent( path, ( p ) -> new FileMeta( p, target.isDirectory() ) );
+                // Check the current lock level
+                // TODO: Negotiate the lock level
+                LockLevel currentLockLevel = meta.getLockLevel( this.nodeKey );
+
+                meta.addLock( this.nodeKey, owner.getLockLevel() );
+                metadataCache.put( path, meta );
+                transactionManager.commit();
 
                 return meta;
             } );
         }
         catch ( InterruptedException e )
         {
-            throw new IOException( "Thread interrupted while retrieving / creating file metadata : " + path, e );
+            e.printStackTrace();
```

Comment:
Log this...it's never ok to just e.printStackTrace(); ...we need context around that error to make sense of it.